pysat/_instrument.py

```diff
@@ -1053,77 +1051,83 @@ def _assign_funcs(self, by_name=False, inst_module=None):
             output_str += 'Number of Orbits: {:d}\n'.format(self.orbits.num)
             output_str += 'Loaded Orbit Number: '
             if self.orbits.current is not None:
-                output_str += '{:d}\n'.format(self.orbits.current)
+                output_str += '{:d}\n\n'.format(self.orbits.current)
             else:
-                output_str += 'None\n'
+                output_str += 'None\n\n'
 
-        output_str += '\nLocal File Statistics' + '\n'
-        output_str += '---------------------' + '\n'
-        output_str += 'Number of files: ' + str(len(self.files.files)) + '\n'
+        # Print the local file information
+        output_str += self.files.__str__()
 
-        if len(self.files.files) > 0:
-            output_str += 'Date Range: '
-            output_str += self.files.files.index[0].strftime('%d %B %Y')
-            output_str += ' --- '
-            output_str += self.files.files.index[-1].strftime('%d %B %Y')
-
-        output_str += '\n\nLoaded Data Statistics' + '\n'
-        output_str += '----------------------' + '\n'
+        # Display loaded data
+        output_str += '\n\nLoaded Data Statistics\n'
+        output_str += '----------------------\n'
         if not self.empty:
+            num_vars = len(self.variables)
+            max_vars = 6
+
             output_str += 'Date: ' + self.date.strftime('%d %B %Y') + '\n'
-            output_str += 'DOY: {:03d}'.format(self.doy) + '\n'
+            output_str += 'DOY: {:03d}\n'.format(self.doy)
             output_str += 'Time range: '
             output_str += self.index[0].strftime('%d %B %Y %H:%M:%S')
             output_str += ' --- '
             output_str += self.index[-1].strftime('%d %B %Y %H:%M:%S\n')
-            output_str += 'Number of Times: ' + str(len(self.index)) + '\n'
-            output_str += 'Number of variables: ' + str(len(self.variables))
-
-            output_str += '\n\nVariable Names:' + '\n'
-            num = len(self.variables) // 3
-            for i in np.arange(num):
-                output_str += self.variables[3 * i].ljust(30)
-                output_str += self.variables[3 * i + 1].ljust(30)
-                output_str += self.variables[3 * i + 2].ljust(30) + '\n'
-            for i in np.arange(len(self.variables) - 3 * num):
-                output_str += self.variables[i + 3 * num].ljust(30)
-            output_str += '\n'
+            output_str += 'Number of Times: {:d}\n'.format(len(self.index))
+            output_str += 'Number of variables: {:d}\n'.format(num_vars)
+
+            if num_vars <= max_vars:
+                output_str += '\nVariable Names:\n'
+                num = len(self.variables) // 3
+                for i in np.arange(num):
+                    output_str += self.variables[3 * i].ljust(30)
+                    output_str += self.variables[3 * i + 1].ljust(30)
+                    output_str += self.variables[3 * i + 2].ljust(30) + '\n'
```

Comment:
These few output string lines are not being tested.

Reply:
And I have no idea why, because I took great pains to make a test that does print out data variables.  Can you try out `test_instrument.py::TestBasics::test_str_w_load_less_data` and see WHY it hates those lines?

Reply:
Suggestion below.